Framework/Muon/test/ApplyMuonDetectorGroupPairingTest.h

```diff
@@ -0,0 +1,631 @@
+    }
+
+    for (auto &&badPeriod : badPeriods) {
+      alg.setProperty("SubtractedPeriods", badPeriod);
+      // This throw comes from MuonProcess
+      TS_ASSERT_THROWS(alg.execute(), std::runtime_error);
+      TS_ASSERT(!alg.isExecuted());
+    }
+  }
+
+  void test_producesOutputWorkspacesInWorkspaceGroup() {
+    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(10, 5);
+    setUpADSWithWorkspace setup(ws);
+
+    ApplyMuonDetectorGroupPairing alg;
+    alg.initialize();
+    setPairAlgorithmProperties(alg, setup.inputWSName, setup.groupWSName);
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+    TS_ASSERT(setup.wsGroup);
+    if (setup.wsGroup) {
+      TS_ASSERT_EQUALS(setup.wsGroup->getNumberOfEntries(), 2);
+    }
+  }
+
+  void test_outputWorkspacesHaveCorrectName() {
+    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(10, 5);
+    setUpADSWithWorkspace setup(ws);
+
+    ApplyMuonDetectorGroupPairing alg;
+    alg.initialize();
+    setPairAlgorithmProperties(alg, setup.inputWSName, setup.groupWSName);
+    alg.execute();
+
+    TS_ASSERT(setup.wsGroup->getItem("inputGroup; Pair; test; Asym; #1"));
+    TS_ASSERT(setup.wsGroup->getItem("inputGroup; Pair; test; Asym; #1_Raw"));
+  }
+
+  void test_workspacePairingHasCorrectAsymmetryValues() {
+    MatrixWorkspace_sptr ws =
+        createAsymmetryWorkspace(10, 10, yDataAsymmetry());
+    setUpADSWithWorkspace setup(ws);
+
+    ApplyMuonDetectorGroupPairing alg;
+    alg.initialize();
+    setPairAlgorithmProperties(alg, setup.inputWSName, setup.groupWSName);
+    alg.execute();
+    auto wsOut = boost::dynamic_pointer_cast<MatrixWorkspace>(
+        setup.wsGroup->getItem("inputGroup; Pair; test; Asym; #1_Raw"));
+
+    // Current behaviour is to convert bin edge x-values to bin centre x-values
+    // (point data) so there is on fewer x-value now.
+    TS_ASSERT_DELTA(wsOut->readX(0)[0], 0.050, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[4], 0.450, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[9], 0.950, 0.001);
+
+    TS_ASSERT_DELTA(wsOut->readY(0)[0], -0.4692, 0.0001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[4], -1.0517, 0.0001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[9], -0.6119, 0.0001);
+
+    // The error calculation as per Issue #5035
+    TS_ASSERT_DELTA(wsOut->readE(0)[0], 0.04212, 0.00001);
+    // TS_ASSERT_DELTA(wsOut->readE(0)[4], -0.00000, 0.00001);
+    TS_ASSERT_DELTA(wsOut->readE(0)[9], 0.06946, 0.00001);
+  }
+
+  void test_timeOffsetShiftsTimeAxisCorrectly() {
+    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(10, 10);
+    setUpADSWithWorkspace setup(ws);
+
+    ApplyMuonDetectorGroupPairing alg;
+    alg.initialize();
+    setPairAlgorithmProperties(alg, setup.inputWSName, setup.groupWSName);
+    alg.setProperty("TimeOffset", 0.2);
+    alg.execute();
+    auto wsOut = boost::dynamic_pointer_cast<MatrixWorkspace>(
+        setup.wsGroup->getItem("inputGroup; Pair; test; Asym; #1_Raw"));
+
```

Comment:
if we have two workspaces: ws1 has no offset and ws2 has an offset of 0.2 does that mean:

`ws1->readX(0)[k] + 0.2 = ws2->readX(0)[k]`

Reply:
Yes, if they begin with the same x-values.

Reply:
making the above change for the `TS_ASSERT` would make the results relative (and thats all that matters for this test)